src/Uno.Extensions.Navigation.Generators/ForceBindingsUpdateGenerationContext.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Uno.Extensions.Generators;
+using Uno.Extensions.Navigation.UI.Controls;
+
+namespace Uno.Extensions.Navigation.Generators;
+
+internal record ForceBindingsUpdateGenerationContext(
+	GeneratorExecutionContext Context,
+
+	// Core types
+	[ContextType("Uno.Extensions.Navigation.IForceBindingsUpdate")] INamedTypeSymbol ForceBindingsUpdateInterface,
+
+	// Attributes
+	[ContextType(typeof(ForceUpdateAttribute))] INamedTypeSymbol UpdateAttribute,
+
+	// General stuff types
+	[ContextType("Microsoft.UI.Xaml.Controls.Page")] INamedTypeSymbol Page)
+{
+	public bool IsGenerationNotDisable(ISymbol symbol)
+		=> IsGenerationEnabled(symbol) ?? true;
+
+	public bool? IsGenerationEnabled(ISymbol symbol)
+		=> symbol.FindAttributeValue<bool>(UpdateAttribute, nameof(ForceUpdateAttribute.IsEnabled), 0) is { isDefined: true } attribute
+			? attribute.value ?? true
+			: null;
+
+	private IImmutableSet<string>? _XBindFiles;
+	private static Regex ClassRegEx = new Regex("x:Class=\"([\\w.]+)\"");
+	private static string xBind = "{x:Bind ";
+
+	private IImmutableSet<string> XBindFiles
+	{
+		get
+		{
+			if (_XBindFiles is null)
+			{
+				var files = ImmutableHashSet.CreateBuilder<string>();
+				foreach (var file in Context.AdditionalFiles)
+				{
+					using (var reader = new StreamReader(file.Path))
+					{
+						string? className = null;
+						var hasXBind = false;
+						while (
+							!reader.EndOfStream &&
+							(className is null ||
+							!hasXBind))
+						{
+							var txt = reader.ReadLine();
```

Comment:
Thinking more, this is not the most efficient code (even for large files case). It is best if we can have a fixed-size buffer that is reused instead of allocating too many small strings. It will be quite more work to do.

Say, if the buffer holds 1024 char, we will search for `{x:Bind` in it, but if the buffer ends with `{x:B` we will need to track that and start searching `ind ` in the next iteration.

Benchmarking would be good to get a feeling of what happens.

NOTE: The benchmark should be done against some .NET Framework version. I don't recall the exact version VS uses, but it won't matter much as long as it's .NET Framework.

Alternatively, we could profile the generator in a real project, instead of benchmarking.

Reply:
@dansiegel / @Youssef1313 this PR is accepting contributions....